joern-cli/frontends/x2cpg/src/main/scala/io/joern/x2cpg/passes/frontend/XTypeRecovery.scala

```diff
@@ -352,7 +354,7 @@ abstract class RecoverForXCompilationUnit[CompilationUnitType <: AstNode](
   protected def associateTypes(symbol: LocalVar, fa: FieldAccess, types: Set[String]): Set[String] = {
     val fieldParents = getFieldParents(fa)
     fa.astChildren.l match {
-      case ::(i: Identifier, _) if isField(i) && fieldParents.nonEmpty =>
+      case (i: Identifier) :: _ if isField(i) && fieldParents.nonEmpty =>
         fieldParents.foreach(fp => globalTable.put(FieldVar(fp, symbol.identifier), types))
```

Comment:
if we only care about the first astChild, then how about something like this?
```scala
    fa.astChildren.headOption.collect {
      case i: Identifier if isField(i) =>
        fieldParents.foreach(fp => globalTable.put(FieldVar(fp, symbol.identifier), types))
    }

```
The `fieldParents.nonEmpty` check doesn't seem necessary either.

Reply:
👍